PollBuddy-Server/backend/routes/polls.js

```diff
@@ -1,64 +1,80 @@
-var createError = require("http-errors");
 var express = require("express");
 var router = express.Router();
 var mongoConnection = require("../modules/mongoConnection.js");
+const Joi = require("joi");
+const {createResponse, validateID} = require("../modules/utils");
 
 router.post("/new", function (req, res) {
-  // Get POST data
-  var jsonContent = req.body;
-
-  // Validate
-  // Name should be present
-  if(!jsonContent.Name) {
-    return res.status(400).send("Error, Name parameter not specified");
+  // Validate request body
+  const schema = Joi.object({
+    Name: Joi.string().min(3).max(30).required()
+  });
+  const validResult = schema.validate(req.body);
+  // invalidate handling
+  if (validResult.error) {
+    return res.status(404).send(createResponse(false, validResult.error.details[0].message));
   }
 
-  // TODO: Need to add more validation like length, characters perhaps, etc.
-
   // Add to DB
-  mongoConnection.getDB().collection("polls").insertOne({Name: jsonContent.Name}, function(err, result) {
+  mongoConnection.getDB().collection("polls").insertOne({Name: validResult.value.Name}, function (err, result) {
     if (err) {
       console.log(err);
       return res.sendStatus(500);
     } else {
-      if(result.result.ok !== 1) {
+      if (result.result.ok !== 1) {
         // Failed to insert for some reason
         return res.sendStatus(500);
       } else {
         // Things seemed to be ok, send result message and ID of inserted object
-        return res.send({ "Result": "Success", "ID": result.insertedId });
+        // return res.send({"Result": "Success", "ID": result.insertedId});
+        return res.send(createResponse(true, null, {ID: result.insertedId}));
       }
     }
   });
 
 });
-router.post("/:id/edit", function (req, res) {
-  var id = new mongoConnection.getMongo().ObjectID(req.params.id);
-  var jsonContent = req.body;
-  if (jsonContent.Action === "Add") {
-    if (jsonContent.Questions !== undefined) {//QUESTION IS AN OBJECT https://docs.google.com/document/d/1kFdjwiE4_POgcTDqXK-bcnz4RAeLG6yaF2RxLzkNDrE/edit
-      mongoConnection.getDB().collection("polls").updateOne({ "_id": id }, { "$addToSet": { Questions: jsonContent.Questions } }, function (err, res) {
-        if (err) {
-          return res.sendStatus(500);
-        }
-      });
-    } else {
-      return res.sendStatus(400);
+router.post("/:id/edit", async (req, res) => {
+  // validate request body
+  const schema = Joi.object({
+    Action: Joi.string().valid("add", "remove").required(), // two mode supported: 'add' or 'remove'
+    Question: Joi.object().keys({
+      QuestionText: Joi.string().min(3).max(512).required(),  // question text capped at 512 words
```

Comment:
Do we have this noted in the wiki?

Reply:
Give that the wiki is pretty much empty I don't think so. 'add' and 'remove' is what's written in the old code, I've just mapped it to use `Joi`. 512 text cap is flexible, I just thought that it's better if we cap everything just incase for a super large post request. I'll make sure to write all of this in the wiki this week.

Reply:
Sure, sounds good